src/lib/nvme_pt_ioctl.c

```diff
@@ -400,14 +455,29 @@ int opal_init_pt(struct sed_device *dev, const char *device_path)
 	memset(opal_dev, 0, sizeof(*opal_dev));
 	dev->priv = opal_dev;
 
-	opal_dev->session.tsn = 0;
-	opal_dev->session.hsn = 0;
+	opal_dev->session.tsn = opal_dev->session.hsn = 0;
+	opal_dev->req_buf = opal_dev->resp_buf = NULL;
 
-	ret = opal_level0_disc_pt(dev);
+	ret = resize_io_buf(opal_dev, MIN_IO_BUFFER_LEN);
 	if (ret)
 		goto init_deinit;
 
-	SEDCLI_DEBUG_PARAM("The device comid is: %u\n", opal_dev->comid);
+	ret = opal_dev_discv(dev);
+	if (ret)
+		goto init_deinit;
+
+	uint64_t max_com_pkt_sz = tper_prop_to_val(dev, MCPS);
```

Comment:
please avoid mixing code with declaration of variable

Reply:
Okay will do it